client/state/plugins/installed/selectors.js

```diff
@@ -105,6 +105,16 @@ export function getPlugins( state, siteIds, pluginFilter ) {
 	return sortBy( pluginList, ( item ) => item.slug.toLowerCase() );
 }
 
+export const getPlugins = createSelector(
+	getPluginsSelector,
+	( state ) => [ state.plugins.installed.plugins ],
+	( state, siteIds, pluginFilter ) => {
+		const pluginIds = Object.keys( state.plugins.installed.plugins );
+
+		return [ siteIds, pluginIds, pluginFilter ].join();
```

Comment:
Did you mean to flatten the array before joining it? That way we won't rely on the engine to perform `.toString()` on the sub-arrays for us.

Also, did you mean to join the items by some character, like a dash for example? Otherwise, the cache key will contain commas, which I'm not sure was intended.

```suggestion
		return [ siteIds, pluginIds, pluginFilter ].flat().join( '-' );
```

Just to elaborate on the need to flat the arrays before joining, see the difference between these two:

```javascript
> [ [ 1, 2 ], [ 3, 4 ], [ 5, 6 ] ].join( '-' )
'1,2-3,4-5,6'

> [ [ 1, 2 ], [ 3, 4 ], [ 5, 6 ] ].flat().join( '-' )
'1-2-3-4-5-6'
```

Reply:
Good idea!